backend/src/main/java/gov/cdc/usds/simplereport/api/heathcheck/BackendAndDatabaseHealthIndicator.java

```diff
@@ -0,0 +1,31 @@
+package gov.cdc.usds.simplereport.api.heathcheck;
+
+import gov.cdc.usds.simplereport.db.repository.FeatureFlagRepository;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.exception.JDBCConnectionException;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+@Component("prod-smoke-test")
+@Slf4j
+@RequiredArgsConstructor
+public class BackendAndDatabaseHealthIndicator implements HealthIndicator {
+  @Getter(AccessLevel.NONE)
+  @Setter(AccessLevel.NONE)
+  private final FeatureFlagRepository _repo;
+
+  @Override
+  public Health health() {
+    try {
+      _repo.findAll();
+      return Health.up().build();
+    } catch (JDBCConnectionException e) {
```

Comment:
Wasn't sure if there was a better error to throw for "database didn't connect". If folks have suggestions happy to change this!

Reply:
I forget if I mentioned this or not while we were pairing, but my thought was to grab the most recent entry from the `databasechangelog` table and report something from that table (maybe the md5checksum?) back to the frontend that might be useful to know.  Ideally some value that is useful but also definitely not sensitive information.

cc @alismx for the idea of having the checksum be visible.

Reply:
Happy to change the db call to the checksum table, but if we want to pass that info back to the frontend, we'll need to configure the [`show-details` flag on the health check config](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints.health) to `always` since we'd be hitting it unauthed. Those details include info about the filepath of the application, state of DB, etc that someone would theoretically be able to see if they found that endpoint.

Don't think there's anything _too_ sensitive here, but since passing data back to the frontend was more of a nice to have, I decided to lean against flipping the flag. If we feel strongly that the checksum is worth having on the frontend / we get a security consult that the extra info isn't an issue, happy to make the relevant changes.

Let me know what you think!